internal/functions/deploy/bundle.go

```diff
@@ -50,8 +50,17 @@ func (b *dockerBundler) Bundle(ctx context.Context, slug, entrypoint, importMap
 	hostOutputPath := filepath.Join(hostOutputDir, "output.eszip")
 	// Create exec command
 	cmd := []string{"bundle", "--entrypoint", utils.ToDockerPath(entrypoint), "--output", utils.ToDockerPath(hostOutputPath)}
+	// Handle import map/config flags based on Deno version
+	// Deno 2: use --config for deno.json files, --import-map for legacy import_map.json
+	// Deno 1: use --import-map for all import map files
 	if len(importMap) > 0 {
-		cmd = append(cmd, "--import-map", utils.ToDockerPath(importMap))
+		if utils.Config.EdgeRuntime.DenoVersion > 1 && function.IsDenoJsonImportMap(importMap) {
```

Comment:
So on closer look, the import map struct is a subset of deno.json file. So we only need to check the deno version here.